scripts/bfg.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+set -ex
+
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null && pwd)"
+cd $DIR
+
+SRC=/tmp/openpilot/
+SRC_ARCHIVE=/tmp/openpilot-release-archive/
+OUT=/tmp/smallpilot/
+
+wget -O /tmp/git-filter-repo https://raw.githubusercontent.com/newren/git-filter-repo/main/git-filter-repo
+chmod +x /tmp/git-filter-repo
+
+if [ ! -d $SRC ]; then
+  git clone --bare --mirror https://github.com/commaai/openpilot.git $SRC
+fi
+
+if [ ! -d $SRC_ARCHIVE ]; then
+  git clone --bare --mirror https://github.com/commaai/openpilot-release-archive.git $SRC_ARCHIVE
+fi
+
+cd $SRC
+
+echo "starting size $(du -sh .)"
+
+git remote update
+
+# the git-filter-repo analysis is bliss - can be found in the repo root/filter-repo/analysis
+/tmp/git-filter-repo --force --analyze
+
+# Add openpilot-release-archive as a remote
+git remote add archive $SRC_ARCHIVE
+
+# Fetch the content from the archive remote
+git fetch archive
+
+# git checkout --track origin/devel
+# WIP: seems we might not need the archive repo at all - since we already have the history of the tags in the main repo
+git checkout tags/v0.7.1
+git checkout -b archive
+
+# rm these so that we don't get an error when adding them as submodules later
+git rm -r cereal opendbc panda
+git commit -m "removed unmergeable files and directories before merge"
+
+# skip-smudge to get rid of some lfs errors that it can't find the reference of some lfs files
+git lfs install --skip-smudge --local
+
+# go to master
+git checkout master
+
+# rebase previous "devel" history
+# WIP - this doesn't complete, it currently errors after ~20 commits rebased
+git rebase -X ours archive
+
+# push to archive repo
+# WIP - push it when rebase is done without errors
+#git push -f --mirror https://github.com/commaai/openpilot-archive.git
+
+rm -rf $OUT
+cp -r $SRC $OUT
+
+cd $OUT
+
+# remove all tags
+# TODO: we need to keep the tags, WIP on how we actually do it
+# git tag -l | xargs git tag -d
+
+# remove all non-master branches
+# TODO: need to see if we "redo" the other branches (except master, master-ci, devel, devel-staging, release3, release3-staging, dashcam3, dashcam3-staging, testing-closet*, hotfix-*)
```

Comment:
would be nice to automatically rebase the other branches on the repo, but guess we can split that out into a second script that runs on the generated repo from here

Reply:
I would also do it separately